improver_tests/cube_combiner/test_CubeCombiner.py

```diff
@@ -205,10 +194,7 @@ def test_mixed_dtypes_overflow(self):
         cubelist = iris.cube.CubeList(
             [self.cube1, self.cube2.copy(np.ones_like(self.cube2.data, dtype=np.int32))]
         )
-        msg = (
-            r"Operation add on types \{dtype\(\'.*\'\)\} results in "
-            r"float64 data which cannot be safely coerced to float32"
-        )
+        msg = "Operation .* results in float64 data"
```

Comment:
The behaviour has very slightly changed, in that this error will now have eg `np.add` instead of `add` in the string.

In terms of why I cut it down though.  I saw this done a while ago in some of the pytests and asked @tjtg about it.  The answer was broadly: we don't really need to verify the whole error string, as we might have legitimate reasons for changing the details in future.  We just want to check enough to identify that it's the error we wanted to raise, and that any important components are there.  In this case I judged that the important components are the fact that we've escalated to 64-bit precision, so I kept that.